app/src/main/java/org/schabi/newpipe/streams/OggFromWebMWriter.java

```diff
@@ -290,6 +317,64 @@ private byte[] makeMetadata() {
         return null;
     }
 
+    /**
+     * This creates a single metadata tag for use in opus metadata headers. It contains the four
+     * byte string length field and includes the string as-is. This cannot be used independently,
+     * but must follow a proper "OpusTags" header.
+     *
+     * @param keyValue A key-value pair in the format "KEY=some value"
+     * @return The binary data of the encoded metadata tag
+     */
+    private static byte[] makeOpusMetadataTag(final String keyValue) {
+        // Ensure the key is uppercase
+        final var delimiterIndex = keyValue.indexOf('=');
+        final var key = keyValue.substring(0, delimiterIndex).toUpperCase();
+        final var value = keyValue.substring(delimiterIndex + 1);
+        final var reconstructedKeyValue = key + "=" + value;
```

Comment:
If you want to use enforce this here, it would be better to add use a map before that. Currently, above happen multiple string concatinations which are later split up again. So I'd prefer to have some sort of Map or better a list of `Pair`s.